burn-tensor/src/tensor/api/base.rs

```diff
@@ -39,6 +39,11 @@ where
         K::shape(&self.primitive)
     }
 
+    /// Returns size of the current tensor.
+    pub fn size(&self) -> usize {
+        self.dims().iter().fold(1, |acc, &dim| acc * dim)
+    }
```

Comment:
There is a similar method `num_elements` in the `Shape` struct, I would use it instead here, but I like the fold implementation, so we may update the `num_elements` function implementation to use `fold` instead of loops. `size` is a bit unclear as a function name, since size often return the `shape` in PyTorch, so we may update the name to `num_elements`.

Reply:
Actually I didn't mean to leave this in- I needed something like this for an earlier implementation I was trying out. We can remove it, rename it, or do whatever honestly 🤣